### PR TITLE
Add `channel` flag to Genjobs

### PR DESCRIPTION
**Pull request by clarketm**
1. Add `--channel` flag to `genjobs` for configuring *slack reporting channel*
2. Make `--bucket` and `--ssh-key-secret` *optional* since they can be set via tuning **new** `default_decoration_configs` field.
3. Regenerate `istio-private` jobs.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: prow/genjobs/cmd/genjobs/main.go [hunk 1]
```diff
@@ -189,31 +183,61 @@ func convertOrgRepoStr(o options, s string) string {
...
+func updateReporterConfig(o options, job *config.JobBase) {
+	if o.channel == "" {
+		return
+	}
+
+	if job.ReporterConfig == nil {
+		job.ReporterConfig = &prowjob.ReporterConfig{}
+	}
+
+	job.ReporterConfig.Slack = &prowjob.SlackReporterConfig{Channel: o.channel}
```
Comment (fejta) [inline prow/genjobs/cmd/genjobs/main.go:238]: It seems like it would be better to have prow handle this transparently. AKA add a reportSlack field

**Review by fejta — APPROVED** [review]
/hold

File: prow/genjobs/cmd/genjobs/main.go (hunk 1 above)
Comment (clarketm) [reply prow/genjobs/cmd/genjobs/main.go:238]: Do you mean something like this https://github.com/kubernetes/test-infra/pull/15159? This flag is only *temporary* until this is supported in Prow config.

Comment (fejta) [reply prow/genjobs/cmd/genjobs/main.go:238]: Fabulous

**Comment by clarketm** [comment]
/hold cancel
I will switch to Prow-specific configuration to achieve this once the feature is available.

Draft PR: https://github.com/istio/test-infra/pull/2064
feature: kubernetes/test-infra#15159

**Pull request merged**